assets/prometheus-k8s/kube-rbac-proxy-secret.yaml

```diff
@@ -9,11 +9,17 @@ metadata:
 stringData:
   config.yaml: |-
     "authorization":
+      "rewrites":
+        "byQueryParameter":
+          "name": "namespace"
+      "static":
+        - "apiGroup": "metrics.k8s.io/v1beta1"
+          "namespace": "{{ .Value }}"
+          "resource": "pods"
+          "resourceRequest": true
```

Comment:
this would actually not verify against a concrete user. the purpose of static authorization is to bind it against a concrete user, see https://github.com/openshift/kube-rbac-proxy/pull/43/files#diff-c6b1500c9d0e568b9d4e4712df03c09f384ce19722c3befbd59847dc4fbfd431R43-R50.

In this case we also need to check against the `system:serviceaccount:openshift-monitoring:prometheus-k8s` user, so the section would need to add:
```
        - user:
            name: system:serviceaccount:openshift-monitoring:prometheus-k8s
```

Reply:
Could you please explain why it needs to be hard coded to this SA. If a user is logged in, isnt that user token used?

Reply:
you don't assert if a user is logged in here. the current declaration is actually a no-op. the purpose of this is use case is to authorize prometheus service accounts only to be able to scrape its own metrics without performing a subject access review against API server. See a more elaborate rationale in https://github.com/deads2k/openshift-enhancements/blob/master/enhancements/monitoring/client-cert-scraping.md